### ISSUE TITLE
Crashing invocation of Objects.requiresNonNullElse is permitted

### ISSUE THREAD

**Issue by mernst** [typetools/checker-framework#3149]
The following code type-checks but should not:
```
    // This should fail typechecks, because it fails at run time.
    System.err.println((Object) requireNonNullElse(null, null));
```

A test case appears in `checker/tests/nullness/ObjectsRequireNonNullElse.java`.

**Comment by mernst** [comment]
Related (or maybe the same issue): #3150

**Comment by wmdietl** [comment]
I agree that this is probably the same reason as #3150 .
A combined test case, using `null` literals and `@Nullable` references would be good.
File `checker/tests/nullness/ObjectsRequireNonNullElse.java` isn't committed yet...

**Comment by mernst** [comment]
The test case (and one for #3150) is added now; sorry for the delay, which was caused by investigation of #3150.

**Comment by mernst** [comment]
Closed in favor of #3150.

**Comment by smillst** [comment]
The JDK annotations for requiresNonNullElse are wrong.  I'll fix them.
